docs/discussion/Ratify v2 design scope.md

```diff
@@ -0,0 +1,165 @@
+    - Eliminate cache usage in external verifiers. This is the simplest solution but may impact the performance.
+    - Do NOT share cache among internal and external verifiers. This is another simple solution but could not bring performance improvement.
+    - Use some IPC mechanism to share cache among internal and external verifiers. For example, we can use mmap to share cache or use socket/pipe to share cache. This is more complex but could bring better performance improvement.
+
+**Priority**: The overall cache refactoring is a high priority task. But different options have different priorities. If we want to choose the first 2 options, it can be a task of priority 1, but if we want to choose the third option, it can be a task of priority 2 as it's more complex.
+
+#### Policy Provider Refactoring
+**Current Limitation**:
+1. The Policy evaluation workflow is coupled within the executor. Updating or adding a new policy provider requires changes to the executor which makes it hard to maintain and extend.
+2. Rego policy provider and Config policy provider return different types of results. This could lead to confusion when users are switching between different policy providers.
+3. There are a few limitations/bugs in config policy that were adrresed by the Rego policy provider. And if we want to keep supporting the config policy provider, we need to make sure it is consistent with the Rego policy provider. [issue-351](https://github.com/ratify-project/ratify/issues/351), [issue-737](https://github.com/ratify-project/ratify/issues/737), [issue-528](https://github.com/ratify-project/ratify/issues/528)
+
+**Proposed Solution**:
+1. Decouple the policy evaluation workflow from the executor. This will make it easier to maintain and extend the policy providers.
+2. Standardize the result format of the policy providers. This will make it easier for users to switch between different policy providers.
+3. Make the rego policy provider as the default policy provider and provide some templates covering most user scenarios.
+4. [optional] Remove the config policy provider or fix the limitations/bugs in the config policy provider.
+
+**Priority**: The overall policy provider refactoring is priority 2. But if we want to fix the limitations/bugs in the config policy provider, it can be a task of priority 3.
+
+#### Configuration Refactoring
+**Current Limitation**:
+1. Ratify CLI does not support all features that are available in the Ratify service. One of the reasons is that the CLI configuration does not support new options, e.g. KeyManagementProvider.
+2. Oras store only supports one auth provider. One of the reasons is that the Oras store configuration does not support multiple auth providers. Related issue: [issue-974](https://github.com/ratify-project/ratify/issues/974).
+
+**Proposed Solution**:
+1. Update the CLI configuration format to be forward-compatible with missing features.
```

Comment:
I would recommend we remove all of the extra non-essential commands in the CLI today. Even the primary `verify` command will need a large overhaul to provide a better configuration experience. Given the complexity of the config file, I think we need a way for user's to compose the config file from the command flags itself.

Reply:
I agree. And we don't have much unit tests for CLI, I think it's a good opportunity for us to redesign the CLI commands (deprecate unused ones and add new ones).